### PR TITLE
webidl binder: define properties with JS accessors

### PR DESCRIPTION
**Pull request by ascorbic**
Currently C++ class and struct properties are defined in JavaScript bindings with `get_foo` and `set_foo` accessor methods. This PR adds support for directly accessing the properties using native JS accessors. For example:

```js
// Current way
myObject.set_foo(1);
console.log(myObject.get_foo());

// After this PR:
myObject.foo = 1;
console.log(myObject.foo);
```

This is more idiomatic JavaScript, and means that the bindings match the IDL correctly. I have left the existing getters and setters in place, so this should be backward-compatible. 

If this PR is accepted I'll update the docs too.

### CODE REVIEW

File: tools/webidl_binder.py [hunk 1]
```diff
@@ -682,8 +682,11 @@ def render_function(class_name, func_name, sigs, return_type, non_pointer, copy,
                     call_content=get_call_content,
                     const=m.getExtendedAttribute('Const'),
                     array_attribute=m.type.isArray())
-
-    if not m.readonly:
+                    
+    if m.readonly:
+      mid_js += [r'''
+    Object.defineProperty(%s.prototype, '%s', {get: %s.prototype.%s }) ''' % (name, attr, name, get_name)]
```
Comment (kripken) [inline tools/webidl_binder.py]: whitespace around the `{ .. }` is unbalanced. please add a space after the `{`

File: tools/webidl_binder.py [hunk 2]
```diff
@@ -697,6 +700,9 @@ def render_function(class_name, func_name, sigs, return_type, non_pointer, copy,
                       call_content=set_call_content,
                       const=m.getExtendedAttribute('Const'),
                       array_attribute=m.type.isArray())
+      mid_js += [r'''
+    Object.defineProperty(%s.prototype, '%s', {get: %s.prototype.%s, set: %s.prototype.%s }) ''' % (name, attr, name, get_name, name, set_name)]
```
Comment (kripken) [inline tools/webidl_binder.py]: also here

**Comment by kripken** [comment]
Oh, also please update the docs, in `site/source/docs/porting/connecting_cpp_and_javascript/WebIDL-Binder.rst`. I see we mention `get_` there, but only in the context of const - would be good to mention setters as well, maybe in a section just for them, and add the new notation docs there.

**Comment by ascorbic** [comment]
Great, thanks. I've made those changes. I've not handled this sort of test before, but I *think* the changes are correct.

**Comment by kripken** [comment]
Thanks @ascorbic! Looks good to me and the tests pass :)

**Pull request merged**